docs/concepts.md

```diff
@@ -0,0 +1,24 @@
+## Concepts
+
+The ksonnet framework is centered around a couple of key concepts.
+
+### Application
+TODO
+
+### Component
+TODO
+
+### Prototype
+TODO
+
+### Parameter
```

Comment:
I know you're going to be updating this in a separate PR, but just wanted to mention that `Environment` should probably be one of these concepts.

Reply:
ahh right, thanks!